google/gax/__init__.py

```diff
@@ -482,3 +491,173 @@ def __next__(self):
         if self._index >= len(self._current):
             self._current = None
         return resource
+
+
+def _from_any(pb_type, any_pb):
+    """Converts an Any protobuf to the specified message type
+
+    Args:
+        pb_type (type): the type of the message that any_pb stores an instance
+            of.
+        any_pb (google.protobuf.any_pb2.Any): the object to be converted.
+
+    Returns:
+        An instance of the pb_type message.
+    """
+    msg = pb_type()
+    if any_pb.Unpack(msg):
+        return msg
+
+    raise TypeError(
+        'Could not convert {} to {}'.format(
+            any_pb.__class__.__name__, pb_type.__name__))
+
+
+def _try_callback(target, clbk):
+    try:
+        clbk(target)
+    except Exception as ex:  # pylint: disable=broad-except
+        _LOG.exception(ex)
+
+
+class _OperationFuture(object):
+    """A Future which polls a service for completion via OperationsClient."""
+
+    def __init__(self, operation, client, result_type, metadata_type,
+                 call_options=None):
+        """Constructor.
+
+        Args:
+            operation (google.longrunning.Operation): the initial long-running
+                operation object.
+            client (google.gapic.longrunning.operations_client.OperationsClient):
+                a client for the long-running operation service.
+            result_type (type): the class type of the result.
+            metadata_type (type): the class type of the metadata.
+            call_options (google.gax.CallOptions, optional): the call options
+                that are used when reloading the operation.
+        """
+        self._last_operation = operation
```

Comment:
Why `_last_operation` instead of just `_operation`? The "last" implies to me that it might hold over time a sequence of operations, rather than a single one whose status is being updated.